File: clients/python/uiform/types/secrets/external_api_keys.py
from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel

from ..ai_models import AIProvider


class ExternalAPIKeyRequest(BaseModel):
    """Request model for creating/updating API keys"""

    provider: AIProvider
    api_key: str


class ExternalAPIKey(BaseModel):
    """Response model for API key information"""

    provider: AIProvider
    is_configured: bool
    last_updated: Optional[datetime]

File: clients/node/src/types/secrets/external_api_keys.ts
import { z } from 'zod';
import { AIProviderSchema } from '../ai_models.js';

export const ExternalAPIKeyRequestSchema = z.object({
  provider: AIProviderSchema,
  api_key: z.string(),
});
export type ExternalAPIKeyRequest = z.infer<typeof ExternalAPIKeyRequestSchema>;

export const ExternalAPIKeySchema = z.object({
  provider: AIProviderSchema,
  is_configured: z.boolean(),
  last_updated: z.string().datetime().nullable().optional(),
});
export type ExternalAPIKey = z.infer<typeof ExternalAPIKeySchema>;